umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso19115_2/tiling_system.clj

```diff
@@ -3,43 +3,46 @@
   (:require [cmr.common.xml.gen :refer :all]
             [cmr.umm-spec.util :refer [char-string]]))
 
-(defn- tiling-system-coding-params
-  "Returns ISO tiling system encoding string parameters (coordinate 1 prefix, coordinate 2 prefix,
-  and range separator) for given tiling system map. The parameters are used by tiling-system-string."
-  [tiling-system]
-  (condp #(.contains %2 %1) (:TilingIdentificationSystemName tiling-system)
-    "CALIPSO" ["o" "p" ","]
-    "MISR"    ["p" "b" "-"]
-    "MODIS"   ["h" "v" "-"]
-    "WRS"     ["p" "r" "-"]
-    ["x" "y" "-"]))
-
 (defn- tiling-system-string
-  "Returns an encoded ISO tiling system coordinate string from the given UMM tiling system. The
-  coordinate 1 and coordinate 2 prefixes and separator may be specified, or else they will be looked
-  up based on the tiling system name."
-  ([tiling-system p1 p2 sep]
-   (let [{{c1-min :MinimumValue
-           c1-max :MaximumValue} :Coordinate1
-           {c2-min :MinimumValue
-            c2-max :MaximumValue} :Coordinate2} tiling-system]
-     (str p1 c1-min
+  "Returns an encoded ISO tiling system coordinate string from the given UMM tiling system."
+  [tiling-system]
+  (let [{{c1-min :MinimumValue
+          c1-max :MaximumValue} :Coordinate1
+         {c2-min :MinimumValue
+          c2-max :MaximumValue} :Coordinate2 } tiling-system]
+      (let [buildString ""]
+        (with-out-str
+          (when c1-min
+            (print "c1-min:" c1-min)
```

Comment:
Yikes. Couple of thoughts on this: 1) I'd pop this string-building out into its own private utility function; 2) I'd rewrite that. The use of `print` with `with-out-str` there is, while undoubtedly creative, pretty far outside the Clojure idiom norms. (It's kind of like `let` block abuse ... turning a chunk of FP code into procedural code.) I'll take a look at it and offer an example of better option ...